ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/googleapis.sheets;
+import ballerina/url;
+import ballerina/http;
+import ballerina/time;
+
+type Table record {
+    map<json>[] cols;
+    RowValues[] rows;
+    int parsedNumHeaders;
+};
+
+type RowValues record {
+    map<json>[] c;
+};
+
+# The client used by the generated persist clients to abstract and
+# execute API calls that are required to perform CRUD operations.
+public client class GoogleSheetsClient {
+
+    private final sheets:Client googleSheetClient;
+    private final http:Client httpClient;
+    private string spreadsheetId;
+    private int sheetId;
+    private string entityName;
+    private string tableName;
+    private string range;
+    private map<SheetFieldMetadata> fieldMetadata;
+    private map<string> dataTypes;
+    private string[] keyFields;
+    private function (string[]) returns stream<record {}, Error?>|Error query;
+    private function (anydata) returns record {}|InvalidKeyError queryOne;
+    private map<function (record {}, string[]) returns record {}[]|error> associationsMethods;
+
+    # Initializes the `GSheetClient`.
+    #
+    # + GSheetClient - The `sheets:Client`, which is used to execute google sheets operations
+    # + httpClient - The `http:Client`, which is used to execute http requests
+    # + metadata - Metadata of the entity
```

Comment:
```suggestion
    # + sheetMetadata - Metadata of the entity
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)